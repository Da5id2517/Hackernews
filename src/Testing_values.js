export const list = [
  {
    title: 'Naslov',
    url: 'https://lemonparty.org',
    author: 'SomeBloke',
    num_comment: 3,
    points: 4,
    objectID: 0
  },
  {
    title: 'Seheron',
    url: 'https://youtube.org',
    author: 'Tseries',
    num_comment: 12,
    points: 3,
    objectID: 1
  }
]

export const blank_func = () => {};
export const test_string = 'test_string';
