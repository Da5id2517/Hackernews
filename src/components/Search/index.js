import React, {Component} from 'react';
import PropTypes from 'prop-types';


export default class Search extends Component {

  componentDidMount() {
    if (this.input) {
      this.input.focus();
    }
  }

  render() {
    const {
      onSearchChange,
      onSubmit,
      searchTerm,
      children,
    } = this.props;

    return(
      <form onSubmit={onSubmit}>
        <input
        type='text'
        onChange={onSearchChange}
        value={searchTerm}
        ref={(node) => {this.input = node;}}
        />
        <button type='submit'>
        {children}
        </button>
      </form>
    );
  }
};

Search.propTypes = {
  onSubmit: PropTypes.func.isRequired,
  onSearchChange: PropTypes.func.isRequired,
  children: PropTypes.node.isRequired,
  searchTerm: PropTypes.string.isRequired,
};
