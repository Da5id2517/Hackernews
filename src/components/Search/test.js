import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Search from './';
import {blank_func, test_string} from '../../Testing_values.js';


describe('Search', () => {
  it('renders okay', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <Search
      onSearchChange={blank_func}
      onSubmit={blank_func}
      value={test_string}
      searchTerm={test_string}>
      {test_string}
      </Search>
      , div
    );
    ReactDOM.unmountComponentAtNode(div);
  })

  it('has valid snap.', () => {
    const component = renderer.create(
      <Search
      onSearchChange={blank_func}
      onSubmit={blank_func}
      value={test_string}
      searchTerm={test_string}>
      {test_string}
      </Search>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })
})
