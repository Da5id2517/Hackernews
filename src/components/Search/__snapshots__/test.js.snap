// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search has valid snap. 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value="test_string"
  />
  <button
    type="submit"
  >
    test_string
  </button>
</form>
`;
